In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import math

In [9]:
import json

In [12]:
data = pd.read_csv("data.csv")


str(data["b'radar_primary_yson_msg'"][0])
# json.loads(str(data["b'radar_primary_yson_msg'"][0]))

"{b'row_step': 960, b'height': 1, b'fields': [{b'name': b'x', b'datatype': 7, b'count': 1, b'offset': 0}, {b'name': b'y', b'datatype': 7, b'count': 1, b'offset': 4}, {b'name': b'z', b'datatype': 7, b'count': 1, b'offset': 8}, {b'name': b'obj_vrel_long', b'datatype': 7, b'count': 1, b'offset': 12}, {b'name': b'obj_lat_speed', b'datatype': 7, b'count': 1, b'offset': 16}, {b'name': b'obj_rcs_value', b'datatype': 7, b'count': 1, b'offset': 20}, {b'name': b'radial_speed_absolute', b'datatype': 7, b'count': 1, b'offset': 24}, {b'name': b'distance_accuracy', b'datatype': 7, b'count': 1, b'offset': 28}, {b'name': b'angle_accuracy', b'datatype': 7, b'count': 1, b'offset': 32}, {b'name': b'pdh0', b'datatype': 7, b'count': 1, b'offset': 36}, {b'name': b'dist_long_rms', b'datatype': 7, b'count': 1, b'offset': 40}, {b'name': b'dist_lat_rms', b'datatype': 7, b'count': 1, b'offset': 44}, {b'name': b'v_long_rms', b'datatype': 7, b'count': 1, b'offset': 48}, {b'name': b'v_lat_rms', b'datatype': 7, b'co

"{b'row_step': 960, 
b'height': 1,
 b'fields': [{b'name': b'x', b'datatype': 7, b'count': 1,  b'offset': 0}, 
            {b'name': b'y', b'datatype': 7, b'count': 1, b'offset': 4},
             {b'name': b'z', b'datatype': 7, b'count': 1, b'offset': 8}, 
             {b'name': b'obj_vrel_long', b'datatype': 7, b'count': 1, b'offset': 12}, 
             {b'name': b'obj_lat_speed', b'datatype': 7, b'count': 1, b'offset': 16}, 
             {b'name': b'obj_rcs_value', b'datatype': 7, b'count': 1, b'offset': 20}, 
             {b'name': b'radial_speed_absolute', b'datatype': 7, b'count': 1, b'offset': 24}, 
             {b'name': b'distance_accuracy', b'datatype': 7, b'count': 1, b'offset': 28}, 
             {b'name': b'angle_accuracy', b'datatype': 7, b'count': 1, b'offset': 32}, 
             {b'name': b'pdh0', b'datatype': 7, b'count': 1, b'offset': 36}, 
             {b'name': b'dist_long_rms', b'datatype': 7, b'count': 1, b'offset': 40}, 
             {b'name': b'dist_lat_rms', b'datatype': 7, b'count': 1, b'offset': 44}, 
             {b'name': b'v_long_rms', b'datatype': 7, b'count': 1, b'offset': 48}, 
             {b'name': b'v_lat_rms', b'datatype': 7, b'count': 1, b'offset': 52}, 
             {b'name': b'dyn_prop', b'datatype': 1, b'count': 1, b'offset': 56}, 
             {b'name': b'range', b'datatype': 1, b'count': 1, b'offset': 57}, 
             {b'name': b'has_quality', b'datatype': 1, b'count': 1, b'offset': 58}, 
             {b'name': b'ambig', b'datatype': 1, b'count': 1, b'offset': 59}, 
             {b'name': b'invalid', b'datatype': 1, b'count': 1, b'offset': 60}], 
b'data': [0, 0, 136, 65, 205, 204, 140, 63, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 192, 63, 112, 80, 3, 186, 205, 204, 204, 61, 165, 71, 176, 59, 0, 0, 128, 62, 170, 241, 130, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 1, 0, 1, 3, 0, 0, 0, 0, 154, 153, 201, 65, 154, 153, 121, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 16, 193, 41, 80, 239, 185, 205, 204, 204, 61, 179, 69, 179, 59, 0, 0, 64, 63, 117, 147, 168, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 0, 0, 12, 66, 51, 51, 227, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 240, 192, 236, 142, 31, 186, 205, 204, 204, 61, 82, 131, 180, 59, 0, 0, 64, 63, 76, 55, 217, 63, 150, 67, 75, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 205, 204, 20, 66, 154, 153, 57, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 160, 192, 58, 166, 1, 186, 205, 204, 204, 61, 65, 230, 176, 59, 0, 0, 64, 63, 117, 147, 168, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 0, 0, 32, 66, 102, 102, 6, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 64, 192, 142, 16, 17, 186, 205, 204, 204, 61, 72, 159, 175, 59, 0, 0, 64, 63, 170, 241, 130, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 102, 102, 42, 66, 0, 0, 96, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 176, 192, 5, 30, 1, 186, 205, 204, 204, 61, 223, 19, 177, 59, 0, 0, 64, 63, 76, 55, 217, 63, 45, 178, 29, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 205, 204, 52, 66, 0, 0, 208, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 192, 192, 149, 84, 26, 186, 205, 204, 204, 61, 254, 0, 179, 59, 0, 0, 64, 63, 76, 55, 217, 63, 150, 67, 75, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 51, 51, 59, 66, 51, 51, 91, 193, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 32, 192, 198, 80, 38, 186, 205, 204, 204, 61, 67, 173, 184, 59, 0, 0, 64, 63, 117, 147, 168, 63, 150, 67, 75, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 51, 51, 67, 66, 102, 102, 170, 193, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 192, 85, 50, 46, 186, 205, 204, 204, 61, 87, 45, 179, 59, 0, 0, 64, 63, 207, 247, 11, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 102, 102, 106, 66, 102, 102, 14, 65, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 176, 192, 8, 21, 240, 185, 205, 204, 204, 61, 194, 51, 179, 59, 0, 0, 64, 63, 186, 73, 52, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 205, 204, 128, 66, 0, 0, 0, 191, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 96, 192, 88, 10, 12, 186, 205, 204, 204, 61, 7, 81, 172, 59, 0, 0, 64, 63, 186, 73, 52, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 154, 153, 139, 66, 205, 204, 108, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 144, 192, 188, 30, 17, 186, 205, 204, 204, 61, 195, 166, 175, 59, 0, 0, 64, 63, 236, 81, 104, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 11, 0, 0, 0, 205, 204, 156, 66, 154, 153, 49, 193, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 191, 27, 31, 26, 186, 205, 204, 204, 61, 202, 242, 178, 59, 0, 0, 64, 63, 207, 247, 11, 64, 117, 147, 168, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 0, 1, 3, 4, 0, 0, 0, 205, 204, 138, 66, 51, 51, 227, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 224, 192, 147, 74, 22, 186, 205, 204, 204, 62, 136, 195, 228, 58, 0, 0, 64, 63, 203, 161, 149, 64, 170, 241, 130, 63, 188, 116, 147, 62, 88, 57, 52, 60, 3, 1, 1, 3, 4, 0, 0, 0, 0, 0, 7, 67, 0, 0, 176, 192, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 193, 235, 197, 15, 186, 205, 204, 204, 62, 136, 195, 228, 58, 0, 0, 64, 63, 205, 204, 192, 64, 203, 161, 149, 64, 188, 116, 147, 62, 88, 57, 52, 60, 3, 1, 1, 3, 4, 0, 0, 0],
b'width': 15,
b'header': {b'seq': 101821, 
            b'stamp': {b'nsecs': 709627782, b'secs': 1737591862}, 
            b'frame_id': b'radar_front_primary'}, 
b'point_step': 64, 
b'is_bigendian': False, 
b'is_dense': True}"